queries/dictionary.py

```diff
@@ -0,0 +1,233 @@
+        nf = "Ekki tókst að fletta upp orðinu {0}".format(icequote(word))
+        q.set_answer(*gen_answer(nf))
+        return None
+
+    # Nothing found
+    if not res or "results" not in res or not len(res["results"]):
+        return not_found()
+
+    # We're only interested in results where fletta string is equal to word being asked about
+    results = [n for n in res["results"] if n.get("fletta") == word and "flid" in n]
+    if not results:
+        return not_found()
+
+    # OK, we have at least one result.
+    # For now, we just naively use the first result
+    first = results[0]
+
+    # Look it up by ID via the REST API
+    url = _WORD_LOOKUP_URL.format(first["flid"])
+    r = query_json_api(url)
+    if not r:
+        return not_found()
+    # print(r)
+
+    items = r.get("items")
+    if not items:
+        return not_found()
+
+    # Results from the islenskordabok.arnastofnun.is API are either
+    # enumerated definitions or a list of explications. We use the
+    # former if available, else the latter
+
+    # Get all enumerated definition IDs ("LIÐUR")
+    expl: List[int] = [i["itid"] for i in items if i.get("teg") == "LIÐUR"]
+    if expl:
+        # Get corresponding "SKÝRING" for each definition ID
+        sk = [
+            i
+            for i in items
+            if (i.get("paritem") in expl) and (i.get("teg") == "SKÝRING")
+        ]
+        df = [i["texti"] for i in sk]
+    else:
+        # Get all definitions ("SKÝRING")
+        df = [i["texti"] for i in items if i.get("teg") == "SKÝRING"]
+        # df = sorted(df, key=len)
+
+    # If only one definition found, things are simple
+    if len(df) == 1:
+        answ = "{0} er {1}".format(icequote(cap_first(word)), icequote(df[0]))
+        voice = answ
+    else:
+        # Otherwise, do some formatting + spell things out nicely for voice synthesis
+        voice = "Orðið {0} getur þýtt: ".format(icequote(word))
+        answ = ""
+        # Generate list of the form "í fyrsta lagi a, í öðru lagi b, ..."
+        for i, x in enumerate(df[: len(_ENUM_WORDS)]):
+            answ += "{0}. {1}\n".format(i + 1, x)
+            enum = "í {0} lagi,".format(_ENUM_WORDS[i])
+            voice += "{0} {1}, ".format(enum, x)
+        answ = answ.rstrip(",.\n").strip() + "."
+        voice = voice.rstrip(",.\n").strip() + "."
+        voice = _clean4voice(voice)
+
+    q.set_answer(dict(answer=answ), answ, voice)
+
+    # Beautify query by placing word being asked about within quote marks
+    bq = q.beautified_query.replace(wnat, icequote(word))
+    q.set_beautified_query(bq)
+    q.set_source(_DICT_SOURCE)
+
+
+def sentence(state, result):
+    """ Called when sentence processing is complete """
+    q: Query = state["query"]
+    if "qtype" in result and "qkey" in result:
+        # Successfully matched a query type
+        try:
+            _answer_dictionary_query(q, result)
+            q.set_qtype(result.qtype)
```

Comment:
I think the protocol is that you don't set `qtype` or `key` if there is an error, i.e. if you don't return an answer to the query.